### PR TITLE
Search Endpoint

### PR DESCRIPTION
**Pull request by s-lakshmanansi**
I tested the functionality of the endpoint and it seems to be working. I also modified the documentation to mark the shift from URL params to BODY params. Finally, I added a git ignore for ourselves (make life easier lol).

Also, feel free to delete this branch if you approve this merge.

### CODE REVIEW

File: api/API_Documentation.md
Comment (ElliotSchumacher) [inline api/API_Documentation.md:314]: I've been looking into it and it looks like having Body params in a GET request is a breach of standards. This can lead to unforseen errors in the future. I think that GET is still the right http method to use but I think we need to find out how to pass the data in the url

File: api/server.js
Comment (ElliotSchumacher) [inline api/server.js:37]: what does multipleStatements do?

File: api/server.js
Comment (ElliotSchumacher) [inline api/server.js:197]: I would like confirmation through testing done by someone that `books.slice()` won't throw an error if `offset` is greater than the number of elements in `books`. Also that `books.slice()` won't throw an error if `offset + resultLength` is greater than the number of elements in `books`

**Review by ElliotSchumacher — COMMENTED** [review]
I was unable to test this code. Lets work this out before merging into master

File: api/server.js
Comment (s-lakshmanansi) [reply api/server.js:37]: Allows for multiple SQL statements to be run in the same db.query() call.

File: api/server.js
Comment (s-lakshmanansi) [reply api/server.js:197]: No errors are caused here (Pythons slice is pretty nice that way), but you can double check if you want.

File: api/API_Documentation.md
Comment (s-lakshmanansi) [reply api/API_Documentation.md:314]: I definitely agree, its a pretty easy change to switch them back to URL params. I was just having trouble testing using url params as everything was optional + arrays became tedious to write.

File: api/server.js
Comment (ElliotSchumacher) [inline api/server.js:308]: My tests fail because of an error in our SQL syntax when I provide a genre search param. We should take a look at this if statement

File: api/server.js
Comment (ElliotSchumacher) [inline api/server.js:194]: Bug: Including both `resultLength` and `offset` returns a list of books that is greater than the `resultLength`.
Debugging results: `offset` and `resultLength` are read as strings by nodejs right now. When we add them, they are added as strings ("1" + "2" = "12")

I'll have time to fix this Saturday but you are welcome to do so yourself if you like beforehand.

Comment (s-lakshmanansi) [reply api/server.js:194]: Ah good catch. It should be fixed. Do we need to do error handling for invalid inputs (ie. front-end passes in a word instead of a number to "offset")?

Comment (ElliotSchumacher) [reply api/server.js:194]: I think we should at some point but not now. I want to get this branch merged ASAP

**Review by ElliotSchumacher — APPROVED** [review]
After thorough testing, I'm confident that this endpoint will behave as expected. I agree with @s-lakshmanansi that authenticating that users are sending proper parameters is important and should be added but not until all other endpoints are functional. I approve the merge into `master`

**Pull request merged**
